src/vsphere_cpi/lib/cloud/vsphere/resources/vm.rb

```diff
@@ -27,20 +27,20 @@ def resource_pool
         properties['resourcePool'].name
       end
 
-      def accessible_datastore_names
-        host_properties['datastore'].map do |store|
-          ds = cloud_searcher.get_properties(
-            store,
-            Vim::Datastore,
-            ['name', 'summary.accessible'],
+      def accessible_datastores
+        Datastore.build_from_client(
+            @client,
+            host_properties['datastore'],
```

Comment:
is there behavior the same as query multiple times? Why were we querying multiple types?

Reply:
why were we querying multiple times, not sure. But behavior is same as both call cloud_searcher.get_properties. get_properties takes care if obj is an array or a single ManagedObject. For this change we need Datastore object  with all properties instead of just 2 properties which were being queried earlier